File: Java/Car.java
package Java;

class Car{
    private Integer id; 
    private String license;
    private Account driver;
    protected Integer passengers;

    public Car(String license, Account driver){
        this.license = license;
        this.driver = driver;
    }

    void printDataCar(){
        if(passengers != null){
            System.out.println("Car license: " + license + " Car driver: " + driver.name + "Mail: " + driver.email + " Passengers: " + passengers);
        }
    }

    public Integer getPassengers(){
        return passengers;
    }

    public void setPassengers(Integer passengers){
        if (passengers == 4){
            this.passengers = passengers;
        }else{
            System.out.println("Necesitas asignar cuatro pasajeros");
        }
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getLicense() {
        return license;
    }

    public void setLicense(String license) {
        this.license = license;
    }

    public Account getDriver() {
        return driver;
    }

    public void setDriver(Account driver) {
        this.driver = driver;
    }

    

}

File: JS/Car.js
class Car{
    constructor(license, driver){
        this.id;
        this.license = license;
        this.driver = driver;
        this.passenger;            
    }

    printDataCar(){
        console.log( this.driver )
        console.log(  this.driver.name )
        console.log(  this.driver.document )             
    }
}

// This was the valid code until ECMAScript 6
// function Car(license, driver){
//     this.id;
//     this.license = license;
//     this.driver = driver;
//     this.passenger;
// }

// Car.prototype.printDataCar = function(){
//     console.log( this.driver )
//     console.log(  this.driver.name )
//     console.log(  this.driver.name )        
// }